### PR TITLE
Make remote upstream name configurable

### LINKED ISSUE #87: There should be a way to configure the upstream remote instead of guessing
The current heuristics to get what the upstream git remote name is is quite weak, as said in #76. Ideally there should be a way for the user to just set it as a neptune-release configuration via `git config` or something.

Probably `git config hub.upstream` could be used as a way to guess too in the process, and after guessing this should be saved in the config for future use.

### PR DESCRIPTION
**Pull request by mathias-baumann-sociomantic**
Fixes #87

Requires #95

### CODE REVIEW

**Comment by mathias-baumann-sociomantic** [comment]
rebased

File: integrationtest/initial_release/main.d [hunk 1]
```diff
@@ -93,6 +93,7 @@ void runNeptune ( RestAPI gh_test )
     tmp.cmd("mkdir -p " ~ GitPath);
     git.cmd("git init");
     git.cmd("git config neptune.upstream " ~ GitRepo);
+    git.cmd("git config neptune.upstream-remote origin");
```
Comment (leandro-lucarella-sociomantic) [inline integrationtest/initial_release/main.d]: What about using `upstreamremote` like [git-hub](https://github.com/sociomantic-tsunami/git-hub/blob/v1.x.x/man.rst#configuration)?

Comment (mathias-baumann-sociomantic) [reply integrationtest/initial_release/main.d]: I think it looks terrible :P but I can change it for consistency

**Comment by mathias-baumann-sociomantic** [comment]
Update:
* rebased
* renamed to `upstreamremote`
* using proper exception catching / handling

File: integrationtest/initial_release/main.d (hunk 1 above)
Comment (leandro-lucarella-sociomantic) [reply integrationtest/initial_release/main.d]: Thanks for making your command uglier :)

**Comment by mathias-baumann-sociomantic** [comment]
Rebased

**Pull request merged**